core/controllers/acl_decorators.py

```diff
@@ -1563,6 +1563,11 @@ def test_can_edit(self, topic_id, *args, **kwargs):
         if not self.user_id:
             raise base.UserFacingExceptions.NotLoggedInException
 
+        try:
+            topic_domain.Topic.require_valid_topic_id(topic_id)
```

Comment:
@aks681 Instead of this try-catch approach should we have `if is_valid_topic_id(..)`? is there any other use of `require_valid_topic_id` which needs this try-catch approach?

Also, for a consistent reason, I wanted to make sure we follow this patter elsewhere in the codebase (for exp/collection/skill etc.)?

Reply:
Hi @DubeySandeep, I think we can change it to an if condition here. I'll go ahead and do this. Thanks!

Reply:
Yeah, if condition sounds fine! Also no, this is the only use of `require_valid_topic_id` afaik.

Reply:
@kevintab95, is this comment resolved?

Reply:
Hi @DubeySandeep, in topic_domain.py I see a general pattern for the `require_valid_*` throwing exceptions. I also see it is the same case with the `utils.require_valid_name` which seems to be used in a bunch of places (for exp, skills, collections). So I think we should stick with the try-catch method here.